app/src/users/admin.py

```diff
@@ -116,6 +116,44 @@ def queryset(self, request, queryset):
         return queryset
 
 
+class UserAccommodationListFilter(admin.SimpleListFilter):
+    title = 'user accommodation'
+    parameter_name = 'user_accommodation'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('no', 'No accommodation'),
+            ('not_continuous', "Not continuous accommodation"),
+            ('partial', "Partial accommodation"),
+            ('full', "Full accommodation"),
+        ]
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        if value == 'no':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=False) &
+                Q(preferences__accommodation_day_2=False) &
+                Q(preferences__accommodation_day_3=False))
+        if value == 'not_continuous':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=True) &
+                Q(preferences__accommodation_day_2=False) &
+                Q(preferences__accommodation_day_3=True))
+        if value == 'partial':
+            return queryset.filter(
+                Q(preferences__accommodation_day_1=True) |
```

Comment:
What was you exact intention for `partial`? This filter suggests, that it means "people that booked at least one night during ZOSIA".